index.js

```diff
@@ -69,16 +72,40 @@ module.exports = function (options = {}) {
 				name += `/${parts.shift()}`;
 			}
 
-			if (parts.length > 0) return;
+			const entry = parts.join('/') || '.';
+
+			let pkg;
+			let dir;
 
 			let search_dir = importer;
 			while (search_dir !== (search_dir = path.dirname(search_dir))) {
-				const dir = path.join(search_dir, 'node_modules', name);
+				dir = path.join(search_dir, 'node_modules', name);
 				const file = path.join(dir, 'package.json');
 				if (fs.existsSync(file)) {
-					const pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
-					if (pkg.svelte) {
-						return path.resolve(dir, pkg.svelte);
+					pkg = JSON.parse(fs.readFileSync(file, 'utf-8'));
+					break;
+				}
+			}
+
+			if (pkg) {
+				// resolve pkg.svelte first
+				if (entry === '.' && pkg.svelte) {
+					return path.resolve(dir, pkg.svelte);
+				}
+
+				const resolved = await this.resolve(importee, importer, { skipSelf: true });
+
```

Comment:
I think we should check here if `svelte` and `default` point different places and we resolved to `default` then we should print a warning

Reply:
This won't happen for the foreseeable future (almost no library will run into this), so I'm in favor of merging now to get it out.